cocos/ui/UIWidget.cpp

```diff
@@ -899,9 +900,38 @@ void Widget::addTouchEventListener(Ref *target, SEL_TouchEvent selector)
     _touchEventSelector = selector;
 }
 
-void Widget::addTouchEventListener(const ccWidgetTouchCallback& callback)
+uint32_t Widget::addTouchEventListener(const ccWidgetTouchCallback &callback)
+{
+	uint32_t key = 0;
+	if (!_touchEventCallbacks.empty())
```

Comment:
please use 4 spaces as indentation, it seems that all of your code use 8 spaces as indentation.